Sources/RabbitMq/Connection.swift

```diff
@@ -1,46 +1,149 @@
 
 public actor Connection {
-    private let config: AMQPConnectionConfiguration
+    private let url: String
     private let eventLoop: EventLoop
+    private let config: AMQPConnectionConfiguration
+    let logger: Logger  // shared to users of Connection
 
     private var channel: AMQPChannel?
     private var connection: AMQPConnection?
+    private let newConsumers: AsyncChannel<RetryingConsumer>
+
+    private var connecting = false
+
+    public var isConnected: Bool {
+        if let conn = self.connection {
+            return conn.isConnected
+        }
+        return false
+    }
 
     public init(
-        eventLoop: EventLoop = MultiThreadedEventLoopGroup(numberOfThreads: 1).next(),
         _ url: String = "",
-        tls: TLSConfiguration = TLSConfiguration.makeClientConfiguration()
+        eventLoop: EventLoop = MultiThreadedEventLoopGroup(numberOfThreads: 1).next(),
+        tls: TLSConfiguration = TLSConfiguration.makeClientConfiguration(),
+        logger: Logger = Logger(label: "\(Connection.self)")
     ) throws {
+        self.url = url
         self.eventLoop = eventLoop
         self.config = try AMQPConnectionConfiguration.init(url: url, tls: tls)
+        self.logger = logger
+
+        self.newConsumers = AsyncChannel<RetryingConsumer>()
     }
 
-    public func isConnected() -> Bool {
-        if let conn = self.connection {
-            return conn.isConnected
+    // Internal use only, for retrying consumers functionality
+    func addRetryingConsumer(consumer: RetryingConsumer) async {
+        await newConsumers.send(consumer)
+    }
+
+    // Method to use to connect without monitoring, will be called when using reuseChannel()
+    public func connect() async throws {
+        if !isConnected && !connecting {
+            connecting = true
+            defer { connecting = false }
+
+            logger.info("Connecting to broker at \(url)")
+            self.connection = try await AMQPConnection.connect(use: self.eventLoop, from: self.config)
+            logger.info("Connected to broker at \(url)")
         }
-        return false
     }
 
-    public func reuseChannel() async throws -> AMQPChannel {
-        guard let channel = self.channel, channel.isOpen else {
-            if self.connection == nil || self.connection!.isConnected {
-                self.connection = try await AMQPConnection.connect(
-                    use: self.eventLoop, from: self.config)
+    private func gracefulCancellableDelay(timeout: Duration) async throws {
+        for await _ in AsyncTimerSequence(interval: timeout, clock: .continuous).cancelOnGracefulShutdown() {
+            break
+        }
+    }
+
+    private func monitorConnection(reconnectionInterval: Duration) async throws {
+        while !Task.isCancelled && !Task.isShuttingDownGracefully {
+            // Ignore if connected
+            if isConnected {
+                try await Task.sleep(for: PollingConnectionSleepInterval)
```

Comment:
I think this should also use `gracefulCancellableDelay`. However, I guess that would mean that you'll have to handle the graceful shutdown in the run loop. Otherwise, it will immediately close the connection.

Reply:
Is this the thing you implemented in the Ascom.Anc.Pg library? I do want an immediate shutdown on graceful shutdown.

Reply:
Okay, I'll try to use it. I forgot that was there...it's been several weeks.

Reply:
Okay, so for here I have some different needs. I don't want to necessarily wait "30 seconds" for a connection. Instead, I want to return immediately if the connection is "connected", or otherwise wait until the timeout. So it's different from the use of `gracefulCancellableDelay` above which just waits to the timeout or gets cancelled on graceful shutdown.

Reply:
I'm not sure we're talking about the same thing here. I meant the sleep for PollingConnectionsSleepInterval, which is used when there is an active connection. This will not exit on a graceful shutdown.

Reply:
Okay, I misunderstood and thought you meant for the whole method! So yes I'll use the method instead of the Task.sleep!!!